invoice/src/main/java/org/killbill/billing/invoice/model/DefaultInvoice.java

```diff
@@ -249,7 +249,9 @@ public BigDecimal getRefundedAmount() {
 
     @Override
     public BigDecimal getBalance() {
-        return hasZeroParentBalance() ? BigDecimal.ZERO : InvoiceCalculatorUtils.computeInvoiceBalance(currency, invoiceItems, payments, isWrittenOff() || isMigrationInvoice());
+        return getStatus().equals(InvoiceStatus.DRAFT) || hasZeroParentBalance() ?
+               BigDecimal.ZERO :
+               InvoiceCalculatorUtils.computeInvoiceBalance(currency, invoiceItems, payments, isWrittenOff() || isMigrationInvoice());
     }
```

Comment:
I am tempted to say it does not matter which kind of invoice it is (i.e current code seems good). We can revisit if needed. 👍